### PR TITLE
Add eslint-config of React Native community

### PR DESCRIPTION
**Pull request by guilleSequeiraWolox**
## Description

I added:

- @react-native-community/eslint-config and apply change that the library said.

- fix problem with imports (import no-unresolved):
  1) remove babelrc.js
  2) add babel.config.js
  3) add jsconfig.json
  4) modify .eslint.rc.js

## Screenshot 

imports

### CODE REVIEW

File: generators/app/templates/src/app/index.ejs [hunk 1]
```diff
@@ -33,9 +33,10 @@ class App extends Component {
     checkAnalyticsTrackingID();
     <%_ } _%>
     <%_ if(features.login) { _%>
-    authSetup(this.props.dispatch);
+    const { dispatch } = this.props;
```
Comment (guidoprinc) [inline generators/app/templates/src/app/index.ejs]: beware that this destruct only happens on login

File: generators/app/tasks/appSetup/babelConfigSetup.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      'module-resolver',
+      {
+        root: ['./src'],
+        alias: {
+          '@app': './src/app',
+          '@assets': './src/app/assets',
+          '@components': './src/app/components',
+          '@config': './src/config',
+          '@constants': './src/constants',
+          '@lottieAssets': './src/lottieAssets',
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/babelConfigSetup.js]: Remove this

File: generators/app/tasks/appSetup/babelConfigSetup.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          '@schemas': './src/schemas',
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/babelConfigSetup.js]: Why this? I think you added some aliases that don't exist in the bootstrap, just keep these:
```
            '@app': './src/app',
            '@components': './src/app/components',
            '@config': './src/config',
            '@constants': './src/constants',
            '@redux': './src/redux',
            '@screens': './src/app/screens',
            '@services': './src/services',
            '@utils': './src/utils'
```

File: generators/app/tasks/appSetup/babelConfigSetup.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const routes = `presets: ['module:metro-react-native-babel-preset'],
+  plugins: [
+    'import-glob',
+    [
+      'module-resolver',
+      {
+        root: ['./src'],
+        alias: {
+          '@app': './src/app',
+          '@assets': './src/app/assets',
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/babelConfigSetup.js]: Remove

File: generators/app/tasks/appSetup/babelConfigSetup.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+          '@components': './src/app/components',
+          '@config': './src/config',
+          '@constants': './src/constants',
+          '@lottieAssets': './src/lottieAssets',
+          '@schemas': './src/schemas',
+          '@screens': './src/app/screens',
+          '@services': './src/services',
+          '@propTypes': './src/propTypes',
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/babelConfigSetup.js]: Remove

File: generators/app/templates/.eslintrc.ejs [hunk 6]
```diff
@@ -1,50 +1,52 @@
...
+    'no-return-assign': 'off',
+    'no-param-reassign': 'off',
+    'no-nested-ternary': 'off',
+    'react/require-default-props': 'off',
+    'react/jsx-filename-extension': ['error', { extensions: ['.js'] }],
+    'react/prop-types': [2, { ignore: ['style', 'children', 'dispatch'] }],
+    'import/order': ['error', { 'newlines-between': 'always' }],
+    'import/prefer-default-export': 'off',
+    'comma-dangle': ['error', 'never'],
+    curly: 'off',
```
Comment (sfernandez11) [inline generators/app/templates/.eslintrc.ejs]: Is it ok that `curly` don't have `' '`?

File: generators/app/templates/jsconfig.ejs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "commonjs",
+    "jsx": "react-native",
+    "allowSyntheticDefaultImports": true,
+    "baseUrl": ".",
+    "paths": {
+      "@app/*": ["./src/app/*"],
```
Comment (sfernandez11) [inline generators/app/templates/jsconfig.ejs]: Same here, you must only keep these:
```
      "@app/*": ["./src/app/*"],
      "@components/*": ["./src/app/components/*"],
      "@config/*": ["./src/config/*"],
      "@constants/*": ["./src/constants/*"],
      "@screens/*": ["./src/app/screens/*"],
      "@services/*": ["./src/services/*"],
      "@redux/*": ["./src/redux/*"],
      "@utils/*": ["./src/utils/*"]
```

File: generators/app/templates/src/redux/store.ejs [hunk 8]
```diff
@@ -6,7 +6,6 @@ import {
   createNavigationReducer
 } from 'react-navigation-redux-helpers';
 import { fetchMiddleware, configureMergeState } from 'redux-recompose';
-import Navigator from '@screens';
```
Comment (sfernandez11) [inline generators/app/templates/src/redux/store.ejs]: Don't delete this line, you must only add a blank line between `
import { fetchMiddleware, configureMergeState } from 'redux-recompose';`  and `import Navigator from '@screens';`

File: generators/app/templates/src/redux/store.ejs [hunk 9]
```diff
@@ -65,4 +65,4 @@ if (__DEV__) enhancers.push(Reactotron.createEnhancer());
 // in dev mode, we'll create the store through Reactotron
 const store = createStore(reducers, compose(...enhancers));
 
-export default store;
+export default store;
```
Comment (sfernandez11) [inline generators/app/templates/src/redux/store.ejs]: Add a blank line below `export default store; `

File: generators/app/templates/src/app/components/AppNavigator/index.js [hunk 10]
```diff
@@ -5,6 +5,7 @@ import { connect } from 'react-redux';
 import { NavigationActions } from 'react-navigation';
 import { createReduxContainer } from 'react-navigation-redux-helpers';
 import { ROOT } from '@constants/platform';
+
```
Comment (guidoprinc) [inline generators/app/templates/src/app/components/AppNavigator/index.js]: why this space?

File: generators/app/templates/src/redux/store.ejs [hunk 11]
```diff
@@ -6,6 +6,7 @@ import {
   createNavigationReducer
 } from 'react-navigation-redux-helpers';
 import { fetchMiddleware, configureMergeState } from 'redux-recompose';
+
```
Comment (guidoprinc) [inline generators/app/templates/src/redux/store.ejs]: ?

**Pull request merged**